linkedlist/doubly/python/DoublyLinkedList.py

```diff
@@ -0,0 +1,257 @@
+        if self.head is None:
+            return "List is Empty!"
+
+        # Check if the head element is the only element in the linkedlist
+        if self.head.next is None:
+            self.head = self.tail = None
+            self.size = 0
+            return
+
+        # Removes reference to head element
+        nodeToRemove = self.head
+
+        newHead = self.head.next
+
+        # Set current head's pointer to None
+        nodeToRemove.next = None
+
+        # Set newHead's previous pointer to None
+        nodeToRemove.previous = None
+
+        # Set newHead as head
+        self.head = newHead
+
+        # Decrement size
+        self.size -= 1
+
+        return nodeToRemove.element
+
+    def peek(self):
+        # Check if list is empty
+        if self.head is None:
+            return "Empty List!"
+
+        return self.head.element
+
+    def removeLast(self):
+        """
+        The `removeLast` method removes the tail of the linkedlist.
+
+        :returns the removed element
+        """
+
+        # Check if linkedlist is empty
+        if self.tail is None:
+            return "No tail present in linkedlist!"
+
+        # Check if the tail is the only element in the list
+
+        if self.tail.previous is None:
+            self.head = None
+            self.tail = None
+            self.size = 0
+            return
+
+        nodeToRemove = self.tail
+        newTail = self.tail.previous
+
+        # Set current tail's previous pointer to None
+        nodeToRemove.previous = None
+        # Set the newTail's next pointer to None
+        newTail.next = None
+        # Set the newTail as tail
+        self.tail = newTail
+
+        # Decrement size
+        self.size -= 1
+
+        return nodeToRemove.element
+
+    def removeAtPosition(self, position):
+        """
+        The `removeAtPosition` method removes a node at the specified postion.
+
+        :param position: position to remove element from
+        :returns the removed element
+        """
+
+        # Check if position is valid
+        if position < 1 or position > self.size:
+            raise IndexError("Invalid position!")
+
+        # Check if element to be removed is the head
+        if position == 0:
+            self.removeFirst()
+            return
+
+        # Check if element to be removed is the tail
+        if position == self.size:
+            self.removeLast()
+            return
+
+        index = 1
+        current = self.head
+
+        # Traverse till we get to position
+        while index != position:
+            index += 1
+            current = current.next
+
+        prev = current.previous
+
+        # Set the previous node's next pointer to the next node
+        prev.next = current.next
+
+        # Set the next node's previous pointer to the previous node
+        current.next.previous = prev
+
+        # Decrement size
+        self.size -= 1
+
+        return current.element
+
+if __name__ == '__main__':
```

Comment:
Is it possible to have something equivalent to Java's `toString()` method that actually prints out all the elements in the linkedlist. Going to add that shortly for the Java version too.

I think it'd be helpful for people to quickly visualize what is in the linkedlist. What do you think? @Youngestdev

Reply:
Yess. I initially thought of that but I was mirroring your work. So, I'll add that now 👍🏾

Reply:
@Youngestdev Added for Java: https://github.com/oyekanmiayo/data-structures-all-langs/blob/main/linkedlist/doubly/java/DoublyLinkedList.java#L283